tests/Microsoft.Fx.Portability.Tests/ApiPortServiceTests.cs

```diff
@@ -4,92 +4,112 @@
-        [MemberData("ApiPortServices")]
-        public async Task ApiPortService_GetAvailableFormatsAsync(ApiPortService apiPortService, EndpointStatus expectedStatus)
-        {
-            var expected = new List<string> { "Json", "HTML", "Excel" };
-
-            var serviceResponse = await apiPortService.GetResultFormatsAsync();
-            var headers = serviceResponse.Headers;
-            var result = serviceResponse.Response;
-
-            Assert.Equal(expectedStatus, headers.Status);
-            Assert.Equal(expected.Count(), result.Count());
-            Assert.Equal(0, expected.Except(result.Select(r => r.DisplayName)).Count());
-        }
-    }
+	public class ApiPortServiceTests
+	{
+		private ApiPortService _apiPortService;
+
+		public ApiPortServiceTests()
+		{
+			var httpMessageHandler = new TestHandler(HttpRequestConverter);
+			var productInformation = new ProductInformation("ApiPort_Tests", typeof(ApiPortServiceTests));
+
+			//Create a fake ApiPortService which uses the TestHandler to send back the response message
+			_apiPortService = new ApiPortService("http://localhost", httpMessageHandler, productInformation);
+		}
+
+		[Fact]
+		public void VerifyParameterChecks()
+		{
+			string endpoint = null;
+			Assert.Throws<ArgumentOutOfRangeException>(() => new ApiPortService(null, new ProductInformation("")));
+			Assert.Throws<ArgumentOutOfRangeException>(() => new ApiPortService(string.Empty, new ProductInformation("")));
+			Assert.Throws<ArgumentOutOfRangeException>(() => new ApiPortService(" \t", new ProductInformation("")));
+		}
+
+		[Fact]
+		public async Task ApiPortService_GetDocIdsWithValidDocIdAsync()
+		{
+			var docIds = new List<string>
+			{
+				"T:System.Console",
+				"M:System.Xml.Serialization.XmlSerializer.Serialize(System.IO.Stream,System.Object)",
+				"M:System.Xml.Serialization.XmlSerializer.Serialize(System.IO.TextWriter,System.Object)",
+				"M:System.Xml.Serialization.XmlSerializer.Serialize(System.Object,System.Xml.Serialization.XmlSerializationWriter)",
+				"M:System.Xml.Serialization.XmlSerializer.Serialize(System.Xml.XmlWriter,System.Object)",
+				"M:System.Xml.Serialization.XmlSerializer.Serialize(System.IO.Stream,System.Object,System.Xml.Serialization.XmlSerializerNamespaces)",
+				"M:System.Xml.Serialization.XmlSerializer.Serialize(System.IO.TextWriter,System.Object,System.Xml.Serialization.XmlSerializerNamespaces)",
+				"M:System.Xml.Serialization.XmlSerializer.Serialize(System.Xml.XmlWriter,System.Object,System.Xml.Serialization.XmlSerializerNamespaces)",
+				"M:System.Xml.Serialization.XmlSerializer.Serialize(System.Xml.XmlWriter,System.Object,System.Xml.Serialization.XmlSerializerNamespaces,System.String)"
+			};
+
+			var serviceResponse = await _apiPortService.QueryDocIdsAsync(docIds);
+			var result = serviceResponse.Response;
+
+			Assert.Equal(docIds.Count(), result.Count());
+			Assert.Equal(0, docIds.Except(result.Select(r => r.Definition.DocId)).Count());
```

Comment:
* I can't tell because a lot of them look the same... but does the test data include docIds that aren't part of the query so we can ensure random ones aren't being returned?
* Is it possible to try a docId that isn't included in the catalog (test data)?